Foundation/Compat/ByteString.hs

```diff
@@ -9,15 +9,19 @@
 {-# LANGUAGE ViewPatterns #-}
 module Foundation.Compat.ByteString
     ( fromByteString
+    , toByteString
     ) where
 
-import Data.ByteString (ByteString)
+import Data.ByteString (ByteString, pack)
 import Data.ByteString.Internal (toForeignPtr)
 import Foundation
-import Foundation.Array
 import Foundation.Array.Internal
 
 -- | Convert a ByteString to a UArray Word8,
 -- without re-allocating or copying anything
 fromByteString :: ByteString -> UArray Word8
 fromByteString = fromForeignPtr . toForeignPtr
+
+-- | Convert a UArray Word8 to a ByteString.
```

Comment:
Sure. Since the array is immutable, that's just about performance and not correctness, right? (And, one would think, it might be fixable). I've added a corresponding comment for all functions.